src/modules/azure.rs

```diff
@@ -25,6 +26,12 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         return None;
     };
 
+    let user_name: Option<UserName> = get_azure_user_name(context);
+    if user_name.is_none() {
+        log::info!("Could not find Azure user name");
+        return None;
```

Comment:
Since this module can replace multiple things now, please handle missing value in `map` and remove the `unwrap` calls there. Since there seems to quite a bit of duplication, please also fetch the user name together with the subscription name.